src/DebugEngineHost/HostLogger.cs

```diff
@@ -3,101 +3,46 @@
-        public delegate void OutputCallback(string outputString);
+        private static ILogChannel s_natvisLogChannel;
+        private static ILogChannel s_engineLogChannel;
 
-        private StreamWriter _streamWriter;
-        private OutputCallback _callback;
-        private readonly object _locker = new object();
+        private static string s_engingLogFile;
 
-        internal HostLogger(StreamWriter streamWriter = null, OutputCallback callback = null)
+        public static void EnableHostLogging(Action<string> callback, LogLevel level = LogLevel.Information)
         {
-            _streamWriter = streamWriter;
-            _callback = callback;
-        }
-
-        public void WriteLine(string line)
-        {
-            lock (_locker)
+            if (s_engineLogChannel == null)
             {
-                if (_streamWriter != null)
-                    _streamWriter.WriteLine(line);
-                _callback?.Invoke(line);
+                s_engineLogChannel = new HostLogChannel(callback, null, level);
             }
         }
 
-        public void Flush()
+        public static void SetEngineLogFile(string logFile)
         {
-            lock (_locker)
-            {
-                if (_streamWriter != null)
-                    _streamWriter.Flush();
-            }
+            s_engingLogFile = logFile;
         }
 
-        public void Close()
+        public static ILogChannel GetEngineLogChannel()
         {
-            lock (_locker)
-            {
-                if (_streamWriter != null)
-                    _streamWriter.Close();
-                _streamWriter = null;
-            }
+            return s_engineLogChannel;
         }
 
-        internal static StreamWriter GetStreamForName(string logFileName, bool throwInUseError)
+        public static ILogChannel GetNatvisLogChannel()
         {
-            if (string.IsNullOrEmpty(logFileName))
-            {
-                return null;
-            }
-            string tempDirectory = Path.GetTempPath();
-            StreamWriter writer = null;
-            if (Path.IsPathRooted(logFileName) || (!string.IsNullOrEmpty(tempDirectory) && Directory.Exists(tempDirectory)))
-            {
-                string filePath = Path.Combine(tempDirectory, logFileName);
-
-                try
-                {
-                    FileStream stream = File.Open(filePath, FileMode.Create, FileAccess.Write, FileShare.Read);
-                    writer = new StreamWriter(stream);
-                }
-                catch (IOException)
-                {
-                    if (throwInUseError)
-                        throw;
-                    // ignore failures from the log being in use by another process
-                }
-            }
-            else
-            {
-                throw new ArgumentOutOfRangeException(nameof(logFileName));
-            }
-            return writer;
+            return s_natvisLogChannel;
```

Comment:
>s_natvisLogChannel [](http://example.com/codeflow?start=19&length=18)

This will always be null. Is that expected?

Reply:
For this PR it will be null. 

With the registry monitor work for VS, this will be set.